src/xunit.assert/Asserts/Sdk/AssertEqualityComparer.cs

```diff
@@ -149,6 +158,36 @@ public bool Equals(T x, T y)
             return dictionaryYKeys.Count == 0;
         }
 
+        bool? CheckIfSetsAreEqual(T x, T y, TypeInfo typeInfo)
+        {
+            if (!IsSet(typeInfo))
+                return null;
+
+            var enumX = x as IEnumerable;
+            var enumY = y as IEnumerable;
+            if (enumX == null || enumY == null)
+                return null;
+
+            var objectEnumX = enumX.Cast<object>();
+            var objectEnumY = enumY.Cast<object>();
+            if (objectEnumX.Count() != objectEnumY.Count())
+                return false;
+
+            var setX = new HashSet<object>(objectEnumX, objectComparer);
```

Comment:
Ok I see what you mean. 
If I can somehow get the Type of T I can maybe do something like this:
```
var comparerType = typeof(Comparer<>).MakeGenericType(typeof(T));
IComparer comparer = (IComparer)Activator.CreateInstance(comparerType);
```
However the hashset c'tor only accepts IEqualityComparer not IComparer. Is that what you had in mind?